java-checks/src/main/java/org/sonar/java/checks/CallOuterPrivateMethodCheck.java

```diff
@@ -92,12 +94,22 @@ private boolean isPrivateMethodOfOuterClass(Symbol symbol) {
     public void checkUsages() {
       for (Symbol methodUsed : usages.elementSet()) {
         if (methodUsed.usages().size() == usages.count(methodUsed)) {
-          MethodTree declaration = (MethodTree) methodUsed.declaration();
-          if (declaration != null) {
-            reportIssue(declaration.simpleName(), "Move this method into \"" + classSymbol.name() + "\".");
-          }
+          reportIssueOnMethod((MethodTree) methodUsed.declaration());
         }
       }
     }
+
+    private void reportIssueOnMethod(@Nullable MethodTree declaration) {
+      if (declaration != null) {
+        String message = "Move this method into ";
+        if (classSymbol.name().isEmpty()) {
```

Comment:
I would really prefer to identify the class as being an anonymous class in the constructor of {{MethodInvocationVisitor}}, with a dedicated {{boolean}} variable, instead of rechecking every time an issue should be reported.